src/internal/connector/exchange/data_collections.go

```diff
@@ -0,0 +1,61 @@
+package exchange
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/pkg/errors"
+
+	"github.com/alcionai/corso/src/internal/connector/graph"
+	"github.com/alcionai/corso/src/internal/data"
+)
+
+// ParseMetadataCollections produces two maps:
+// 1- paths: folderID->filePath, used to look up previous folder pathing
+// in case of a name change or relocation.
+// 2- deltas: folderID->deltaToken, used to look up previous delta token
+// retrievals.
+func ParseMetadataCollections(
+	ctx context.Context,
+	colls []data.Collection,
+) (map[string]string, map[string]string, error) {
+	var (
+		paths  = map[string]string{}
+		deltas = map[string]string{}
+	)
+
+	for _, coll := range colls {
+		items := coll.Items()
+
+		for {
+			var breakLoop bool
```

Comment:
this is personal preference, but if something is used for a flag I tend to explicitly set it to false just so I don't have to take the extra step to remember what the default value is
```suggestion
		breakLoop := false
```

Reply:
Technically that's non-idiomatic go.  General wisdom suggests that any declaration to the zeroth value of a (non-pointer) variable should use the `var` creation.  Personally, I take that side as well.